### DISCUSSION TITLE
Native way to track the list of search terms or queries that returned no results

### DISCUSSION THREAD

**Discussion by hermesalvesbr** [meilisearch/product#646] [Feedback & Feature Proposal]
It would be interesting to know what Meilisearch hasn't found yet, so I could improve the indexed content and better serve the user who used the search box.

I suggest a functionality to **log** searches that do _not return any results_, words (phrases) that are not found in the search.

**Comment by macraig** [comment]
Hi @hermesalvesbr , apologies for the late reply!

Could you expand on how this would help you improve the indexed content? Can you think of an example of how you improved this lately (perhaps with a workaround)? I'd love to understand your use case better to file it as an idea for the future :)

**Comment by hermesalvesbr** [comment]
1. Identifying Missing Content: When users search for terms that return no results, it indicates that the indexed content might not cover certain topics or keywords that are relevant to your user base. By logging these searches, you can identify what users are looking for but not finding, and then work to include these missing terms in your indexed content.
2. Refining Synonyms and Variants: Users might use synonyms, alternate spellings, or related terms that your current search system doesn't recognize. By analyzing the searches that return no results, you can discover synonyms or variants that users commonly use, which would allow you to create synonym mappings and improve the search experience.
3. Content Gap Analysis: By collecting data on unsuccessful searches, you can identify topics or subjects that are trending or becoming important to your users but aren't well-covered in your indexed content. This can guide your content creation strategy to ensure you're meeting your users' needs.
4. Enhancing User Experience: When users encounter searches with no results, it can be frustrating. By actively working to reduce the number of unsuccessful searches, you can improve the overall user experience and increase user satisfaction with your platform.
5. Optimizing Search Algorithms: Analyzing unsuccessful searches might reveal patterns in user behavior or language that can help you optimize your search algorithms. For instance, if users often search with long phrases that yield no results, it might indicate a need for improved phrase matching algorithms.

**Reply by macraig** [reply]
Thank you for expanding @hermesalvesbr ! I'll leave this discussion open for other users to upvote and add to it.
